firstvoices/backend/views/sites_views.py

```diff
@@ -93,6 +108,35 @@ def list(self, request, *args, **kwargs):
 
         return Response(data)
 
+    def update(self, request, *args, **kwargs):
+        request_data = request.data
+        site_slug = kwargs["slug"]
+        logo_id = request_data["logo"]
+        banner_image_id = request_data["banner_image"]
+        banner_video_id = request_data["banner_video"]
+
+        # Verifying if the media belongs to the site
+        if logo_id and not verify_media_source(site_slug, "image", logo_id):
+            raise ValidationError("Logo supplied is not from the specified site.")
+
+        if banner_image_id and not verify_media_source(
+            site_slug, "image", banner_image_id
+        ):
+            raise ValidationError("Image supplied is not from the specified site.")
+
+        if banner_video_id and not verify_media_source(
+            site_slug, "video", banner_video_id
+        ):
+            raise ValidationError("Video supplied is not from the specified site.")
+
+        return super().update(request, args, kwargs)
+
+    def get_serializer_class(self):
+        if self.action in ["list", "retrieve"]:
+            return SiteDetailSerializer
+        else:
+            return SiteDetailWriteSerializer
```

Comment:
Would it make sense to just always return the write serializer, since it is the same for the retrieve action anyway?

Reply:
The retrieve action returns the `SiteDetailSerializer`. 
The write serializer only returns the UUIDs in place of logo, bannerImage and bannerVideo instead of returning the complete objects as in the `SiteDetailSerializer`. If that works for the frontend, we can switch all 3 methods (list, retrieve and update) to the `SiteDetailWriteSerializer`.

Reply:
I thought the SiteDetailWriteSerializer was returning the SiteDetailSerializer for responses, and was only different on the input?

Reply:
Oh right. My bad. Adding the change.